### PR TITLE
Add ContentList component

### PR DESCRIPTION
**Pull request by mirceacr**
https://github.com/spartansystems/component-booster/issues/13

Why:

* We need a Content List that takes in items from an array
  and renders them in a `<UL>` or `<OL>`, with or without a
  title, with custom width for the items, wrapped in a `<DIV>`
  with a custom class.

This change addresses the need by:

* Merge remote-tracking branch `origin/feature/hack-page-and-webpack`
  into `feature-contentlist-component` so Webpack paths work and we
  can use `components/ComponentName` instead of `../ComponentName`,
  when importing a component inside another one.
* Add the component.
* Add the story for the component.
* Add basic styling.
* Create tests for different cases of use.
* Remove the option for a custom class from the component and
  storybook, and replace it with a static class `.listWrap`.

### CODE REVIEW

File: universal/components/ContentList/component.js [hunk 1]
```diff
@@ -47,35 +46,19 @@ function renderList ({
...
+  itemWidth
+}: ContentListProps) => {
   return (
-    <SectionFullWidth>
-      <SectionContainer>
-        {renderTitle({hasTitle, title})}
-        {renderList({bulletList, items, itemWidth})}
-      </SectionContainer>
-    </SectionFullWidth>
+    <div className={customClass}>
```
Comment (andandandandrew) [inline universal/components/ContentList/component.js]: Can we have `className={customClass}` only render into that div if that prop has something input for it?

Comment (andandandandrew) [inline universal/components/ContentList/component.js]: Or just remove it. I'm trying to think of a use case where you'd need this, but 99% of the time it is going to be nested in another component, ie SectionContainer.

I'm also thinking about that markup. Not sure that the `<ul> or <ol>` needs to be nested in a `<div>` container.

File: universal/components/ContentList/stories.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  .addDecorator(withKnobs)
+  .add('with props', () => {
+    return (
+      <WithNotes notes={notes}>
+        <ContentList
+          hasTitle={boolean('Do you need a title for this list?', props.hasTitle, props.hasTitle[1])}
+          title={text('Title of list:', props.title)}
+          bulletList={boolean('Is this a bulleted list?', props.bulletList, props.bulletList[1])}
+          itemWidth={select('how wide do you want each list item?', props.itemWidth)}
+          customClass={text('CSS class for the div wrapper:', props.customClass)}
```
Comment (andandandandrew) [inline universal/components/ContentList/stories.js]: If you get rid of the `className={customClass}` prop remember to nix this too.

File: universal/components/ContentList/component.js (hunk 1 above)
Comment (mirceacr) [inline universal/components/ContentList/component.js]: ok, will remove them and let you know after I update the PR.

File: universal/components/ContentList/styles.css [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    list-style-position: outside;
+    padding-left: .25rem;
+  }
+}
+
+/* li widths */
+.itemWidth {
+  margin: 0 0 .25em 0;
+  padding: 0 0;
+  font-family: MonLight;
```
Comment (andandandandrew) [inline universal/components/ContentList/styles.css]: We need to nix this line. `MonLight` is a variable we used on Thrive.

**Comment by mirceacr** [comment]
@andandandandrew updated!

File: universal/components/ContentList/component.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  items: Array<string>,
+  itemWidth: 'full' | 'half' | 'quarter' | 'third'
+}
+
+// Logic for renderTitle
+function renderTitle ({
+  hasTitle,
+  title
+}: { hasTitle: boolean, title: string }) {
+  if (hasTitle) {
```
Comment (bbrock25) [inline universal/components/ContentList/component.js]: we should avoid using a boolean here along with the property. if we just check to see if the title exists, that should be sufficient.

you can remove this entire function `renderTitle` and replace it with only the following in the body of the actual component:
```javascript
return (
<div className={styles.listWrap}>
   {title && <h1>{title}</h1>}
   {renderList({bulletList, items, itemWidth})}
  </div>
)

File: universal/components/ContentList/component.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  itemWidth
+}: ContentListProps) => {
+  if (!items) {
+    throw new Error('you need to supply items for the list')
+  }
+
+  return (
+    <div className={styles.listWrap}>
+      {renderTitle({hasTitle, title})}
+      {renderList({bulletList, items, itemWidth})}
```
Comment (bbrock25) [inline universal/components/ContentList/component.js]: I'd just pull both of these inline and not worry about the render functions, they're overcomplicating the matter for now. 

```javascript
return (
<div className={styles.listWrap}>
   {title && <h3>{title}</h3>}
   {items.length > 0 && <ul>{items.map( i => <li>{i}</li>)}}
  </div>
)
```

**Pull request merged**